ql/src/java/org/apache/hadoop/hive/ql/parse/rewrite/MergeRewriter.java

```diff
@@ -178,8 +178,8 @@ protected static class MergeWhenClauseSqlGenerator implements MergeStatement.Mer
     @Override
     public void appendWhenNotMatchedInsertClause(MergeStatement.InsertClause insertClause) {
       sqlGenerator.append("INSERT INTO ").append(mergeStatement.getTargetName());
-      if (insertClause.getColumnListText() != null) {
-        sqlGenerator.append(' ').append(insertClause.getColumnListText());
+      if (insertClause.getColumnList() != null) {
+        sqlGenerator.append(' ').append(String.join(",",insertClause.getColumnList()));
```

Comment:
nit, space `String.join(", ", insertClause.getColumnList())`

Reply:
should we create a helper method 'toString(List<String> value)'?

Reply:
Added space. 
I think the helper method is `String.join` here. Column list is joined only once. I don't have strong opinion about this.

Reply:
ok